src/controller.cpp

```diff
@@ -586,7 +586,8 @@ geometry_msgs::Twist Controller::update(const double target_x_vel,
   // to de-accelerate and thus avoid abrupt velocity changes at the end of the trajectory
   // The sample time plays an important role on how good these estimates are.
   // Thus We add a distance to the end phase distance estimation depending on the sample time
-  if (current_x_vel > target_end_x_vel)
+  if ((sign(current_target_x_vel_) > 0 && current_x_vel > target_end_x_vel) ||
+      (sign(current_target_x_vel_) < 0 && current_x_vel < target_end_x_vel))
```

Comment:
Should this be `|| (sign(current_target_x_vel_) < 0 && current_x_vel < -target_end_x_vel))`

Maybe it's better to do this with absolute values

Reply:
And also, comparisons like `sign(...) < 0` is equivalent to `... < 0`

Reply:
> Should this be `|| (sign(current_target_x_vel_) < 0 && current_x_vel < -target_end_x_vel))`

No, we check the case where we have negative velocity and our current velocity is 'too big'. `target_end_x_vel` is signed, so we can have a plan to arrive with `target_end_x_vel=-1.0` and if we drive -2.0 now, we should 'decelerate'.